testing/integration/upgrade_test.go

```diff
@@ -327,6 +340,37 @@ func (s *StandaloneUpgradeTestSuite) TestUpgradeStandaloneElasticAgentToSnapshot
 	// s.Require().Equal(expectedAgentHashAfterUpgrade, version.Commit, "agent commit hash changed after upgrade")
 }
 
+func extractCommitHashFromArtifact(t *testing.T, ctx context.Context, artifactVersion *version.ParsedSemVer, agentProject tools.Project) string {
+	tmpDownloadDir := t.TempDir()
+
+	operatingSystem := runtime.GOOS
+	architecture := runtime.GOARCH
+	suffix, err := atesting.GetPackageSuffix(operatingSystem, architecture)
+	require.NoErrorf(t, err, "error determining suffix for OS %q and arch %q", operatingSystem, architecture)
+	prefix := fmt.Sprintf("elastic-agent-%s", artifactVersion.VersionWithPrerelease())
+	pkgName := fmt.Sprintf("%s-%s", prefix, suffix)
+	require.Containsf(t, agentProject.Packages, pkgName, "Artifact API response does not contain pkg %s", pkgName)
+	artifactFilePath := filepath.Join(tmpDownloadDir, pkgName)
+	err = atesting.DownloadPackage(ctx, t, http.DefaultClient, agentProject.Packages[pkgName].URL, artifactFilePath)
+	require.NoError(t, err, "error downloading package")
+	err = atesting.ExtractArtifact(t, artifactFilePath, tmpDownloadDir)
+	require.NoError(t, err, "error extracting artifact")
+	extractedDir := ""
+	entries, err := os.ReadDir(tmpDownloadDir)
+	require.NoError(t, err, "unable to read temp directory")
+	for _, e := range entries {
+		if e.IsDir() && strings.HasPrefix(e.Name(), "elastic-agent-") {
+			extractedDir = e.Name()
+			break
+		}
+	}
+	hashFilePath := filepath.Join(tmpDownloadDir, extractedDir, ".build_hash.txt")
```

Comment:
I wonder if you could simplify some of this by using https://pkg.go.dev/path/filepath#Glob?

Reply:
thought about it, then I discovered we have multiple `.build_hash.txt` in the package and I didn't want to risk picking the wrong one :stuck_out_tongue:

Reply:
> we have multiple `.build_hash.txt` in the package

I think you are referring to this?

```
$ tar tzf elastic-agent-8.9.0-SNAPSHOT-darwin-aarch64.tar.gz| grep build
elastic-agent-8.9.0-SNAPSHOT-darwin-aarch64/.build_hash.txt
elastic-agent-8.9.0-SNAPSHOT-darwin-aarch64/data/elastic-agent-cb1258/components/.build_hash.txt
```

Couldn't you use `filepath.Glob(filepath.Join(tmpDownloadDir,  "elastic-agent-*", "build_hash.txt"))`?

Just seems like a lot of boilerplate that could be avoided.  🤷

Reply:
I was trying with `tmpDownloadDir/*/.build_hash.txt` which was a couple of levels away from getting the beats `.build_hash.txt` but your solution seems definitely better, will change it